### PR TITLE
[examples.sat.hillclimb] Fix required failure probability per round

### LINKED ISSUE #52: Verify that failure probability epsilon is used correctly
Cade et al. propose picking `eps = eps_overall / n` as a heuristic (see 4.3.1 in Cade et al.), we should do the same.

- [x] Adjust calculations in simple hillclimber
- [x] Verify quantum statistics anew (there is a discrepancy with the plots in their paper, but we understand why)
- [x] Check steep hillclimber statistics

### PR DESCRIPTION
**Pull request by anurudhp**
fixes #52

### CODE REVIEW

File: examples/sat/hillclimber.py [hunk 1]
```diff
@@ -47,6 +47,12 @@ def hill_climber(
     x = rng.choice([-1, 1], n)
     w = inst.weight(x)
 
+    # error probability per hillclimb step, assuming a maximum of `n` rounds (see 4.3.1 in https://arxiv.org/pdf/2203.04975.pdf)
+    if error is not None:
```
Comment (catch22) [inline examples/sat/hillclimber.py:51]: you could simplify this as
```
if error is not None:
  error /= n
```

**Review by catch22 — CHANGES_REQUESTED** [review]
This PR aims to fix #52, but I believe the following two items still need work before this PR can be merged:
- [x] Verify quantum statistics anew (compare with existing reference data and papers)
- [ ] Check steep hillclimber statistics

File: examples/sat/hillclimber.py (hunk 1 above)
Comment (timbittner) [reply examples/sat/hillclimber.py:51]: did this

**Comment by timbittner** [comment]
As far as I can tell steep works if the quantum factor is set to `1`, see attached image.
hillclimber_speed_quantumfactor1

**Comment by catch22** [comment]
(See Slack)

**Pull request merged**